### PR TITLE
remove deprecated modules netflow, fb_apache and azure

### PR DESCRIPTION
**Pull request by kaisecheng**
## Release notes

[rn:skip]
## What does this PR do?

removed deprecated modules netflow, fb_apache and azure

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files (and/or docker env variables)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [ ]

## Related issues

- Closes: https://github.com/elastic/logstash/issues/16357
- relates: https://github.com/elastic/logstash/pull/16548

### CODE REVIEW

**Comment by kaisecheng** [comment]
[doc](https://www.elastic.co/guide/en/logstash/current/logstash-modules.html) has instructions on using modules with netflow as example. The removal of doc will be addressed in another pr.

**Comment by yaauie** [comment]
TODO: companion PR for `8.x` that emits an entry on the deprecation logger when either of these deprecated modules is activated.

Looking at the implementation, the best hook for this is likely `LogStash::Modules::Scaffold#with_settings`, which is only called when the plugin is being activated.

Or possibly by hard-coding a list of them in `LogStash::Config::ModulesCommon::pipeline_configs` to compare against its lvar `specified_and_available_names`.

File: x-pack/modules/README.md [hunk 1]
```diff
@@ -1,99 +0,0 @@
-# Module settings and structure
```
Comment (jsvd) [inline x-pack/modules/README.md:1]: given we're not yet removing all modules, maybe we can lead the READMEs intact for now

**Review by jsvd — APPROVED** [review]
Other than maybe restoring the modules/README.md and x-pack/modules/README.md, LGTM

Follow up PRs:
1. `8.x` deprecation entry when these modules are used
2. `main` cleanup docs related to netflow, fb_apache and azure modules

**Pull request merged**